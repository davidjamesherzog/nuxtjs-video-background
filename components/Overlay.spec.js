import { mount } from '@vue/test-utils';
import Overlay from '@/components/Overlay.vue';

describe('Overlay', () => {
  describe('templates', () => {
    test('is a Vue instance', () => {
      const wrapper = mount(Overlay);
      expect(wrapper.vm).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });

    test('should create overlay', () => {
      const wrapper = mount(Overlay);
      const overlay = wrapper.find('[data-overlay]');

      expect(overlay.exists()).toBeTruthy();
      expect(overlay.classes()).toContain('overlay');
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
