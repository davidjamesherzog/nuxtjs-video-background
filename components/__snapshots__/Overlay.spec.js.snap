// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay templates is a Vue instance 1`] = `
<div
  class="overlay"
  data-overlay=""
/>
`;

exports[`Overlay templates should create overlay 1`] = `
<div
  class="overlay"
  data-overlay=""
/>
`;
